File: fronted/src/api/workflow.ts
import request from '@/api/request'

function wsID(): string {
  return localStorage.getItem('active_ws_id') || ''
}

export interface WorkflowRequestVo {
  id: string
  createdAt: string
  workspaceId: string
  requestedBy: string
  requestedByName: string
  requestedByEmail: string
  resourceType: string
  resourceName: string
  action: string
  status: 'pending' | 'approved' | 'rejected' | 'executed' | 'failed'
  reviewedBy?: string
  reviewedByName?: string
  reviewedAt?: string
  reviewNote?: string
  executedAt?: string
  errorMessage?: string
}

export interface WorkflowListParams {
  resourceType?: string
  action?: string
  status?: string
  page?: number
  pageSize?: number
}

export const listWorkflowRequests = (params?: WorkflowListParams) =>
  request.get(`/workspaces/${wsID()}/workflow-requests`, params ?? {})

export const getWorkflowRequest = (id: string) =>
  request.get(`/workspaces/${wsID()}/workflow-requests/${id}`, {})

export const approveWorkflowRequest = (id: string, note?: string) =>
  request.post(`/workspaces/${wsID()}/workflow-requests/${id}/approve`, { note })

export const rejectWorkflowRequest = (id: string, note?: string) =>
  request.post(`/workspaces/${wsID()}/workflow-requests/${id}/reject`, { note })

File: internal/server/vo/workflow.go
package vo

// InvitePreviewVo is returned by the public invite preview endpoint.
type InvitePreviewVo struct {
	Email         string `json:"email"`
	WorkspaceID   string `json:"workspaceId"`
	WorkspaceName string `json:"workspaceName"`
	InviterName   string `json:"inviterName"`
	InviterEmail  string `json:"inviterEmail"`
	Role          string `json:"role"`
	ExpiresAt     string `json:"expiresAt"`
	Status        string `json:"status"`
}

// WorkflowRequestVo is the HTTP response shape for a workflow approval request.
type WorkflowRequestVo struct {
	ID          string `json:"id"`
	CreatedAt   string `json:"createdAt"`
	WorkspaceID string `json:"workspaceId"`

	RequestedBy      string `json:"requestedBy"`
	RequestedByName  string `json:"requestedByName"`
	RequestedByEmail string `json:"requestedByEmail"`

	ResourceType string `json:"resourceType"`
	ResourceName string `json:"resourceName"`
	Action       string `json:"action"`

	Status string `json:"status"`

	ReviewedBy     string  `json:"reviewedBy,omitempty"`
	ReviewedByName string  `json:"reviewedByName,omitempty"`
	ReviewedAt     *string `json:"reviewedAt,omitempty"`
	ReviewNote     string  `json:"reviewNote,omitempty"`

	ExecutedAt   *string `json:"executedAt,omitempty"`
	ErrorMessage string  `json:"errorMessage,omitempty"`
}
